### DISCUSSION TITLE
Type of response = octokit.request(...); response.status hardcoded to 200

### DISCUSSION THREAD

**Discussion by paulober** [octokit/octokit.js#2609] [Q&A]
If type of the HTTP Status code in the response of `octokit.request` is hardcoded to 200, what's the point of including this useless property? And how can I get the real HTTP Status code to compare it against `const HTTP_STATUS_NOT_MODIFIED = 304;`?

Thanks in advance...

**Comment by gr2m** [comment]
> HTTP Status code in the response of `octokit.request` is hardcoded to 200

it's not

> HTTP Status code to compare it against `const HTTP_STATUS_NOT_MODIFIED = 304`

304 status is thrown as error so you can catch it and implement your own caching logic:
https://github.com/octokit/request.js/blob/dca880ca80f0d90c794bff68c1c3739d055d9dd3/src/fetch-wrapper.ts#L90-L100

You can check with `if (error.response?.status === 304) { ... }`

**Reply by paulober** [reply]
@gr2m Thabks for the response. I implemented it and it does work. Though when I don't use a personal access token it get a 500 error back. But if I do the same with curl, it also works without a PAT.
Don't know where the issue could be.

**Reply by gr2m** [reply]
Hmm that might warrant a separate issue. 500 errors should never happen.

You can use `.request.endpoint(...)` to receive an object with generic request options. It will not include the authorization header, but you can inspect all other request options to make sure that they match what you send with curl

**Reply by paulober** [reply]
I still get following error (thought I think this happens less):
```json
{
  name: "HttpError",
  status: 500,
  Stack: "HttpError: Unexpected end of JSON input\n    at /Users/Developer/project/.yarn/cache/@octokit-request-npm-8.1.2-e350fab15b-769710b846.zip/node_modules/@octokit/request/dist-src/fetch-wrapper.js:97:11\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at getReleases (/Users/Developer/project/src/utils/githubREST.mts:113:10)\n    at Main._getHtmlForWebview (/Users/Developer/project/src/main.mts:916:29)\n    at Main._update (/Users/Developer/project/src/webview/newProjectPanel.mts:844:18)",
  request: {
    method: "GET",
    url: "https://api.github.com/repos/raspberrypi/pico-sdk/releases",
    headers: {
      accept: "application/vnd.github.v3+json",
      "user-agent": "octokit-rest.js/20.0.2 octokit-core.js/5.0.1 Node.js/18.15.0 (darwin; arm64)",
      "x-github-api-version": "2022-11-28",
      "if-none-match": "W/\"d01db009b9f05f6c43086fe08b87d0849e8bb486988fb459960df7a0de539631\"",
    },
    request: {
      hook: function () { [native code] },
    },
  },
  vslsStack: [
    {
    },
    {
    },
    {
    },
    {
    },
    {
    },
  ],
}
```

**Reply by gr2m** [reply]
I have not seen this before. If you can create a reproducible test case I can have a look, otherwise I won't be able to help I'm afraid

**Reply by paulober** [reply]
Ok, I may supply a test case so this issue can later be resolved. But for now I bypassed the issue by implementing my own requests with https module in nodejs.
